ogr/ogrsf_frmts/geojson/ogrgeojsonwriter.cpp

```diff
@@ -871,10 +871,41 @@ json_object *OGRGeoJSONWriteAttributes(OGRFeature *poFeature,
             const size_t nLen = strlen(pszStr);
             poObjProp = nullptr;
             if (eSubType == OFSTJSON ||
-                (pszStr[0] == '{' && pszStr[nLen - 1] == '}') ||
-                (pszStr[0] == '[' && pszStr[nLen - 1] == ']'))
+                ((pszStr[0] == '{' && pszStr[nLen - 1] == '}') ||
+                 (pszStr[0] == '[' && pszStr[nLen - 1] == ']')))
             {
-                OGRJSonParse(pszStr, &poObjProp, false);
+                if (poFeature->GetNativeMediaType() &&
+                    strcmp(poFeature->GetNativeMediaType(),
+                           "application/vnd.geo+json") == 0 &&
+                    poFeature->GetNativeData())
+                {
+                    json_object *poNativeObjProp = nullptr;
+                    OGRJSonParse(poFeature->GetNativeData(), &poNativeObjProp,
+                                 false);
+                    if (json_object *poProperties = OGRGeoJSONFindMemberByName(
+                            poNativeObjProp, "properties"))
+                    {
+                        if (json_object *poProperty =
+                                OGRGeoJSONFindMemberByName(
+                                    poProperties, poFieldDefn->GetNameRef()))
+                        {
+                            if (strcmp(json_object_get_string(poProperty),
+                                       pszStr) == 0)
+                            {
+                                poObjProp = poProperty;
```

Comment:
to increase refcont, due to the below json_object_put()
```suggestion
                                poObjProp = json_object_get(poProperty);
```